pykoopman/koopman.py

```diff
@@ -105,24 +114,40 @@ def fit(self, x, u=None, dt=1):
 
         if u is None:
             self.n_control_features_ = 0
-        elif not isinstance(self.regressor, DMDc):
+        elif not isinstance(self.regressor, DMDc) and not isinstance(
+            self.regressor, EDMDc
+        ):
             raise ValueError(
-                "Control input u was passed, but self.regressor is not DMDc"
+                "Control input u was passed, but self.regressor is not DMDc or EDMDc"
+            )
+
+        regressor = self.regressor
+        if y is not None:
+            regressor = EnsembleBaseRegressor(
+                regressor=self.regressor,
+                func=self.observables.transform,
+                inverse_func=self.observables.inverse,
             )
 
         steps = [
             ("observables", self.observables),
-            ("regressor", self.regressor),
+            ("regressor", regressor),
         ]
         self.model = Pipeline(steps)
 
         action = "ignore" if self.quiet else "default"
         with catch_warnings():
             filterwarnings(action, category=UserWarning)
-            if u is None:
+            if u is None and y is None:
                 self.model.fit(x)
             else:
-                self.model.fit(x, u)
+                self.model.fit(x, y, regressor__u=u)
```

Comment:
regressor__u?

Reply:
model is a scikit Pipeline object and this is how one can specify step-specific variables, here for regressor step